outputs/graphite/graphite.go

```diff
@@ -0,0 +1,122 @@
+package graphite
+
+import (
+	"errors"
+	"fmt"
+	"github.com/influxdb/influxdb/client/v2"
+	"github.com/influxdb/telegraf/outputs"
+	"log"
+	"math/rand"
+	"net"
+	"strings"
+	"time"
+)
+
+type Graphite struct {
+	// URL is only for backwards compatability
+	Servers []string
+	Prefix  string
+	Timeout int
+	conns   []net.Conn
+}
+
+var sampleConfig = `
+    [[outputs.graphite]]
+        # TCP raw endpoint for your graphite instance.
+        servers = ["mygraphiteserver:2003"] # required
+        # Prefix metrics name
+        prefix = "ncs.myk8snamespace.nmspgw" # default ""
+        # Timeout
+        timeout = 3 # default 2
+
+`
+
+func (g *Graphite) Connect() error {
+	// Set default values
+	if g.Timeout <= 0 {
+		g.Timeout = 2
+	}
+
+	// Get Connections
+	var conns []net.Conn
+	for _, server := range g.Servers {
+		conn, err := net.DialTimeout("tcp", server, time.Duration(g.Timeout)*time.Second)
+		if err == nil {
+			conns = append(conns, conn)
+		}
+	}
+	g.conns = conns
+	return nil
+}
+
+func (g *Graphite) Close() error {
+	// Closing all connections
+	for _, conn := range g.conns {
+		conn.Close()
+	}
+	return nil
+}
+
+func (g *Graphite) SampleConfig() string {
+	return sampleConfig
+}
+
+func (g *Graphite) Description() string {
+	return "Configuration for Graphite server to send metrics to using TCP raw protocol"
+}
+
+// Choose a random server in the cluster to write to until a successful write
+// occurs, logging each unsuccessful. If all servers fail, return error.
+func (g *Graphite) Write(points []*client.Point) error {
+	// Prepare data
+	var bp []string
+	for _, point := range points {
+		// Get name
+		names := strings.SplitN(point.Name(), "_", 2)
+		var name string
+		if len(names) > 1 {
+			name = names[1]
+		} else {
+			name = point.Name()
+		}
+		// Convert UnixNano to Unix timestamps
+		timestamp := point.UnixNano() / 1000000000
+
+		graphitePoint := fmt.Sprintf("%s.%s %s %d\n",
+			strings.Replace(point.Tags()["host"], ".", "_", -1),
+			strings.Replace(name, ".", "_", -1),
+			point.Fields()["value"],
```

Comment:
@sparrc  
This is not too overkill when you have only one KEY ?
What do you think if I do

if nb_keys == 1 {
    graphitePoint = PREFIX.HOSTNAME.NAME VALUE time 
} else {
        graphitePoint = PREFIX.HOSTNAME.NAME.KEY VALUE time
}
Or maybe
graphitePoint = PREFIX.HOSTNAME.KEY VALUE time ???

Or maybe if key == name